src/views/CloneRepo.vue

```diff
@@ -1,20 +1,110 @@
 <template>
   <ViewWrapper title="Clone down a repository">
-    <div>
+    <label for="repo-url-input">
+      <strong>
+        Enter url:
+      </strong>
+    </label>
+    <div class="repo-url-select">
+      <input
+        v-model="url"
+        id="repo-url-input"
+        class="repo-url-input"
+        placeholder="Enter url"
+      />
+      <button :disabled="invalidUrl" @click="clone">Clone</button>
+    </div>
+    <p v-if="invalidUrl">
+      <strong><em>
+        That repo already exists.
+      </em></strong>
+    </p>
+    <div class="folder-picker">
+      <ReposFolderPicker />
     </div>
   </ViewWrapper>
 </template>
 
 <script>
+import { mapActions, mapState } from 'pinia';
+
+import { alertsStore } from '@/stores/alerts.js';
+import { gitCloneStore } from '@/stores/gitClone.js';
+import { reposStore } from '@/stores/repos.js';
+
+import ReposFolderPicker from '@/components/appsettings/ReposFolderPicker.vue';
 import ViewWrapper from '@/views/ViewWrapper.vue';
 
 export default {
   name: 'CloneRepo',
   components: {
-    ViewWrapper
+    ViewWrapper,
+    ReposFolderPicker
+  },
+  data: function () {
+    return {
+      url: ''
+    };
+  },
+  methods: {
+    clone: async function () {
+      if (!this.url) {
+        return;
+      }
+
+      if (!this.reposFolder) {
+        return;
+      }
+
+      const repoPath = this.reposFolder + '/' + this.repoName;
+
+      try {
+        await this.cloneRepo(this.url, repoPath);
+        this.addRepoToList(repoPath);
+        this.url = '';
+        this.$router.push({ name: 'commits' });
+      } catch (error) {
+        this.addErrorAlert('Error cloning repository.', error);
+      }
+    },
+    ...mapActions(alertsStore, [
+      'addErrorAlert'
+    ]),
+    ...mapActions(reposStore, [
+      'addRepoToList'
+    ]),
+    ...mapActions(gitCloneStore, [
+      'cloneRepo'
+    ])
+  },
+  computed: {
+    invalidUrl: function () {
+      for (let i = 0; i < this.reposList.length; i++) {
+        if (this.reposList[i].title === this.repoName) {
+          console.log(this.reposList[i].title, this.repoName);
+          return true;
+        }
+      }
+      return false;
+    },
```

Comment:
> We actually already have an array of all the sidebar names as strings that we can check against as a Getter in the Repos store.
 >  ``` 
 >  repoAlreadyExists: function () {
 >     return this.sortedRepoPaths.includes(this.repoName);
 >  }
 > ```

This won't work with sorted paths values structured like: `['/Users/user/repos/sporkle']`

Reply:
I guess you'd need to use the destinationPath rather than the repoName when checking against it. That is safer as the filepath for the repo is used as the unique "ID".

Reply:
In the case that they clone the same repo to a different folder should we handle this differently on the sidebar? 

![image](https://github.com/sporkle-app/sporkle/assets/2582391/ac4b65fc-3187-4fe4-a9b4-6e19dd648d15)

Reply:
I think it's okay, it's unlikely that you'd have two repos with the same name, especially since you can just rename the folder locally. Also on hover it shows the full path. Not an edge case I'm concerned with.